### PR TITLE
added translations from BE for categorie

### PR DESCRIPTION
**Pull request by domnelkin**
## Description

Added a possibility to get translations for Categories from GraphQL.

link backend: https://github.com/ita-social-projects/horondi_client_be/pull/920

#### Screenshots

 |         Updated          |
 | :----------------------: |
| image |

### Checklist

- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date admin and back-end part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

File: src/context/categories/categories-context.js [hunk 1]
```diff
@@ -9,7 +9,14 @@ const CategoriesContextProvider = ({ children }) => {
   const [categories, setCategories] = useState([]);
 
   const { loading, error } = useQuery(getAllCategoriesQuery, {
-    onCompleted: (data) => setCategories(data.getAllCategories.items)
+    onCompleted: (data) => {
+      const categoriesList = data.getAllCategories.items;
+      categoriesList.forEach((categorie) => {
```
Comment (maarekua) [inline src/context/categories/categories-context.js:14]: It is not a good idea to mutate an argument incoming in function, in our situation it is a category. Better use the map and apply a new array for setCategories.

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/pages/home/categories-list/categories-list.js [hunk 2]
```diff
@@ -18,19 +18,19 @@ import './categories-carousel.css';
 const CategoriesList = () => {
   const styles = useStyles();
   const { categories } = useContext(CategoriesContext);
-  const { i18n } = useTranslation();
+  const { t, i18n } = useTranslation();
   const language = i18n.language === 'ua' ? 0 : 1;
```
[the remark this answers is not in the document]
Comment (domnelkin) [reply src/pages/home/categories-list/categories-list.js]: Thanks. Useless field removed.

File: src/context/categories/categories-context.js (hunk 1 above)
Comment (domnelkin) [reply src/context/categories/categories-context.js]: Got it. Refactored.

File: src/context/categories/constants.js [hunk 3]
```diff
@@ -14,6 +14,7 @@ export const getAllCategoriesQuery = gql`
           large
         }
         available
+        translations_key
```
Comment (vasyaukr) [inline src/context/categories/constants.js:17]: use camelCase instead of snake_case

Comment (domnelkin) [reply src/context/categories/constants.js:17]: It's general approach in the project. And this field wrote in the same way in all the other components.

Comment (domnelkin) [reply src/context/categories/constants.js]: Changed

**Review by AnastasiaKhukhra — APPROVED** [review]

**Review by olga-gchak — APPROVED** [review]

**Pull request merged**